src/as_execution/mod.rs

```diff
@@ -15,81 +15,198 @@ use crate::settings::max_number_of_pages;
 use crate::tunable_memory::LimitingTunables;
 use crate::GasCosts;
 
-pub(crate) use common::*;
 pub(crate) use context::*;
 pub(crate) use error::*;
 
 #[derive(Clone)]
 pub enum RuntimeModule {
-    ASModule((ASModule, Engine)),
+    ASModule(ASModule),
 }
 
 impl RuntimeModule {
-    /// TODO: Dispatch module creation corresponding to the first bytecode byte
+    /// Dispatch module creation corresponding to the first bytecode byte
     ///
-    /// * (1) target AssemblyScript
+    /// * (1) TODO: target AssemblyScript (remove ident)
     /// * (2) TODO: target X
-    /// * (_) target AssemblyScript and use the full bytecode
-    pub fn new(bytecode: &[u8], limit: u64, gas_costs: GasCosts) -> Result<Self> {
+    /// * (_) target AssemblyScript
+    pub fn new(
+        bytecode: &[u8],
+        limit: u64,
+        gas_costs: GasCosts,
+        cache_compatible: bool,
```

Comment:
what does this argument mean ?

Reply:
This is from a previous version, resolving